nansat/mappers/mapper_sentinel1_l1.py

```diff
@@ -415,125 +368,288 @@ def __init__(self, filename, gdalDataset, gdalMetadata,
+        gcps = Mapper.create_gcps(annotation_data['longitude'],
+                                  annotation_data['latitude'],
+                                  annotation_data['height'],
+                                  annotation_data['pixel'],
+                                  annotation_data['line'])
         self.dataset.SetGCPs(gcps, NSR().wkt)
-        self.dataset.SetMetadataItem('time_coverage_start',
-                                     doc.findall(".//*[{http://www.esa.int/safe/sentinel-1.0}startTime]")[0][0].text)
-        self.dataset.SetMetadataItem('time_coverage_end',
-                                     doc.findall(".//*[{http://www.esa.int/safe/sentinel-1.0}stopTime]")[0][0].text)
-        self.dataset.SetMetadataItem('platform', json.dumps(pti.get_gcmd_platform(missionName)))
+        # set metadata
+        self.dataset.SetMetadataItem('time_coverage_start', manifest_data['time_coverage_start'])
+        self.dataset.SetMetadataItem('time_coverage_end', manifest_data['time_coverage_end'])
+        platform_name = manifest_data['platform_familyName'] + manifest_data['platform_number']
+        self.dataset.SetMetadataItem('platform', json.dumps(pti.get_gcmd_platform(platform_name)))
         self.dataset.SetMetadataItem('instrument', json.dumps(pti.get_gcmd_instrument('SAR')))
-        self.dataset.SetMetadataItem('Entry Title', missionName + ' SAR')
+        self.dataset.SetMetadataItem('Entry Title', platform_name + ' SAR')
         self.dataset.SetMetadataItem('Data Center', 'ESA/EO')
         self.dataset.SetMetadataItem('ISO Topic Category', 'Oceans')
-        self.dataset.SetMetadataItem('Summary', missionName + ' SAR data')
-
+        self.dataset.SetMetadataItem('Summary', platform_name + ' SAR data')
+        self.dataset.FlushCache()
 
+    @staticmethod
+    def correct_geolocation_data(data, max_height=5):
+        """ Correct lon/lat values in geolocation data for points high above ground (incorrect)
+
+        First, Lon/Lat are converted to X/Y in meters. Second, Pixel coordinates are approximated
+        by 2nd order polynomial of input X,Y,Z. Third, this polynomial is used to calculate new
+        Pixel coordinate for ocean surface (Z=0). Fourth, a temporary VRT from original X, Y, Line
+        and corrected Z,Pixel coordinates is created. Fifth, temporary VRT is used for converting
+        original Pixel/Line coordinates into correct Lon/Lat
+
+        Parameters
+        ----------
+        data : dict
+            Original geolocation data from read_geolocation_lut()
+        max_height : int
+            Maximum afordable height
+
+        Returns
+        -------
+        data : dict
+            Corrected geolocation data with new longitude and latitude
+
+        """
+        # don't correct geolocation data if only few points are affected
+        if (data['height'] > max_height).sum() < 10:
+            return data
+
+        # convert XY from degrees to meters in stereographic projection
```

Comment:
This will work for other areas than the poles as well, right?

Reply:
Yes, it is not 'polar stereographic', but simply 'stereographic' projection.